app/src/main/java/com/jvoye/tasky/core/database/RoomLocalTaskyItemDataSource.kt

```diff
@@ -42,23 +42,23 @@ class RoomLocalTaskyItemDataSource(
             }
 
         return combine(tasks, reminders, events) { tasks, reminders, events ->
-            tasks + reminders + events
+            (tasks + reminders + events).sortedBy { it.time }
         }
 
     }
 
     override suspend fun getTaskyItem(taskyType: TaskyType, taskyItemId: String): TaskyItem {
         return when(taskyType) {
             TaskyType.TASK -> {
-                taskDao.getTask(taskyItemId).toTaskyItem()
+                taskDao.getTask(taskyItemId)?.toTaskyItem() ?: throw Exception("Task not found")
```

Comment:
This is one of those places where it’s worth stepping back a bit. Right now the code is throwing exceptions when an item isn’t found locally, but in a sync-heavy flow it’s totally possible that an ID won’t be there anymore -  and that’s not really an “exceptional” case. If we bubble that up as a crash, probably fine in a dev or testing scenario where you want the crash to surface quickly, but in production it’s risky -  you’ll hurt stability metrics (like Play Store’s crash-free users) and frustrate users with sudden exits.

Probably a better fit here is to surface this as a proper state or event that the UI can react to (e.g. “item no longer available, not found, etc”), instead of letting it blow up. That way we still communicate the problem clearly, but without risking app stability.